Extproc crash when converting from converse to openai chat response for ToolCall

*Description*:
Extproc crashes when parsing the toolUse converse response from AWS bedrock, there are two issues:
- The code assumes that the text and toolUse are in the same content block which is wrong.
- There could be only one openai message response choices as AWS bedrock does not support N(multiple choices) defined on OpenAI chat completion request schema.

*Repro steps*:
- Send a prompt with following tool config 
```json
{
    "tools": [
        {
            "toolSpec": {
                "name": "get_weather",
                "description": "Get weather of a location.",
                "inputSchema": {
                  "json": {
                    "type": "object",
                    "properties": {
                        "city": {
                            "type": "string",
                            "description": "City of the location"
                        },
                        "state": {
                            "type": "string",
                            "description": "State of the location"
                        }
                    },
                    "required": ["city", "state"]
                }
              }
            }
        }
    ]
}
```

*Logs*:
```
time=2025-02-18T20:58:07.705Z level=INFO msg="Processing request" path=/v1/chat/completions model=us.anthropic.claude-3-5-sonnet-20240620-v1:0
time=2025-02-18T20:58:07.705Z level=INFO msg="Selected backend" backend=aws-bedrock
bedrock output message from converse: 2
bedrock output message: i=0, text=Certainly! I can help you find out the current weather in New York City. To get this information, I'll use the get_weather function. Let me fetch that data for you right away., toolResult=<nil>, toolUse=<nil>
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbe1ec8]
goroutine 30 [running]:
github.com/envoyproxy/ai-gateway/internal/extproc/translator.(*openAIToAWSBedrockTranslatorV1ChatCompletion).ResponseBody(0xc00045a870, 0xc00042c720, {0x13460e0, 0xc00031c1b0}, 0x1)
	/home/runner/work/ai-gateway/ai-gateway/internal/extproc/translator/openai_awsbedrock.go:585 +0x548
github.com/envoyproxy/ai-gateway/internal/extproc.(*chatCompletionProcessor).ProcessResponseBody(0xc00045a370, {0x1357bc8?, 0x1cb0020?}, 0xc0003320a0)
```

--- a/internal/extproc/translator/openai_awsbedrock.go
+++ b/internal/extproc/translator/openai_awsbedrock.go
@@ -562,7 +562,7 @@ func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) ResponseBody(respHeaders 
 
 	openAIResp := openai.ChatCompletionResponse{
 		Object:  "chat.completion",
-		Choices: make([]openai.ChatCompletionResponseChoice, 0, len(bedrockResp.Output.Message.Content)),
+		Choices: make([]openai.ChatCompletionResponseChoice, 0),
 	}
 	// Convert token usage.
 	if bedrockResp.Usage != nil {
@@ -577,20 +577,25 @@ func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) ResponseBody(respHeaders 
 			CompletionTokens: bedrockResp.Usage.OutputTokens,
 		}
 	}
-	for i, output := range bedrockResp.Output.Message.Content {
-		choice := openai.ChatCompletionResponseChoice{
-			Index: (int64)(i),
-			Message: openai.ChatCompletionResponseChoiceMessage{
-				Content: output.Text,
-				Role:    bedrockResp.Output.Message.Role,
-			},
-			FinishReason: o.bedrockStopReasonToOpenAIStopReason(bedrockResp.StopReason),
-		}
+	// AWS Bedrock does not support N(multiple choices) > 0, so there could be only one choice.
+	choice := openai.ChatCompletionResponseChoice{
+		Index: (int64)(0),
+		Message: openai.ChatCompletionResponseChoiceMessage{
+			Role: bedrockResp.Output.Message.Role,
+		},
+		FinishReason: o.bedrockStopReasonToOpenAIStopReason(bedrockResp.StopReason),
+	}
+	for _, output := range bedrockResp.Output.Message.Content {
 		if toolCall := o.bedrockToolUseToOpenAICalls(output.ToolUse); toolCall != nil {
 			choice.Message.ToolCalls = []openai.ChatCompletionMessageToolCallParam{*toolCall}
+		} else if output.Text != nil {
+			// For the converse response the assumption is that there is only one text content block, we take the first one.
+			if choice.Message.Content == nil {
+				choice.Message.Content = output.Text
+			}
 		}
-		openAIResp.Choices = append(openAIResp.Choices, choice)
 	}
+	openAIResp.Choices = append(openAIResp.Choices, choice)
 
 	mut.Body, err = json.Marshal(openAIResp)
 	if err != nil {

--- a/internal/extproc/translator/openai_awsbedrock_test.go
+++ b/internal/extproc/translator/openai_awsbedrock_test.go
@@ -889,8 +889,6 @@ func TestOpenAIToAWSBedrockTranslatorV1ChatCompletion_ResponseBody(t *testing.T)
 						Role: "assistant",
 						Content: []*awsbedrock.ContentBlock{
 							{Text: ptr.To("response")},
-							{Text: ptr.To("from")},
-							{Text: ptr.To("assistant")},
 						},
 					},
 				},
@@ -907,22 +905,6 @@ func TestOpenAIToAWSBedrockTranslatorV1ChatCompletion_ResponseBody(t *testing.T)
 						Index: 0,
 						Message: openai.ChatCompletionResponseChoiceMessage{
 							Content: ptr.To("response"),
-							Role:    "assistant",
-						},
-						FinishReason: openai.ChatCompletionChoicesFinishReasonStop,
-					},
-					{
-						Index: 1,
-						Message: openai.ChatCompletionResponseChoiceMessage{
-							Content: ptr.To("from"),
-							Role:    "assistant",
-						},
-						FinishReason: openai.ChatCompletionChoicesFinishReasonStop,
-					},
-					{
-						Index: 2,
-						Message: openai.ChatCompletionResponseChoiceMessage{
-							Content: ptr.To("assistant"),
 							Role:    "assistant",
 						},
 						FinishReason: openai.ChatCompletionChoicesFinishReasonStop,
@@ -974,9 +956,12 @@ func TestOpenAIToAWSBedrockTranslatorV1ChatCompletion_ResponseBody(t *testing.T)
 				Output: &awsbedrock.ConverseOutput{
 					Message: awsbedrock.Message{
 						Role: awsbedrock.ConversationRoleAssistant,
+						// Text and ToolUse are sent in two different content blocks for AWS Bedrock, OpenAI merges them in one message.
 						Content: []*awsbedrock.ContentBlock{
 							{
 								Text: ptr.To("response"),
+							},
+							{
 								ToolUse: &awsbedrock.ToolUseBlock{
 									Name:      "exec_python_code",
 									ToolUseID: "call_6g7a",

--- a/internal/extproc/translator/openai_awsbedrock_test.go
+++ b/internal/extproc/translator/openai_awsbedrock_test.go
@@ -889,6 +889,8 @@ func TestOpenAIToAWSBedrockTranslatorV1ChatCompletion_ResponseBody(t *testing.T)
 						Role: "assistant",
 						Content: []*awsbedrock.ContentBlock{
 							{Text: ptr.To("response")},
+							{Text: ptr.To("from")},
+							{Text: ptr.To("assistant")},
 						},
 					},
 				},

--- a/tests/extproc/testupstream_test.go
+++ b/tests/extproc/testupstream_test.go
@@ -120,7 +120,7 @@ func TestWithTestUpstream(t *testing.T) {
 			responseBody:    `{"output":{"message":{"content":[{"text":"response"},{"text":"from"},{"text":"assistant"}],"role":"assistant"}},"stopReason":null,"usage":{"inputTokens":10,"outputTokens":20,"totalTokens":30}}`,
 			expRequestBody:  `{"inferenceConfig":{},"messages":[],"modelId":null,"system":[{"text":"You are a chatbot."}]}`,
 			expStatus:       http.StatusOK,
-			expResponseBody: `{"choices":[{"finish_reason":"stop","index":0,"logprobs":{},"message":{"content":"response","role":"assistant"}},{"finish_reason":"stop","index":1,"logprobs":{},"message":{"content":"from","role":"assistant"}},{"finish_reason":"stop","index":2,"logprobs":{},"message":{"content":"assistant","role":"assistant"}}],"object":"chat.completion","usage":{"completion_tokens":20,"prompt_tokens":10,"total_tokens":30}}`,
+			expResponseBody: `{"choices":[{"finish_reason":"stop","index":0,"logprobs":{},"message":{"content":"response","role":"assistant"}}],"object":"chat.completion","usage":{"completion_tokens":20,"prompt_tokens":10,"total_tokens":30}}`,
 		},
 		{
 			name:            "openai - /v1/chat/completions",

